lib/server.js

```diff
@@ -779,420 +733,306 @@ Server.prototype.toString = function toString() {
 
     str += sprintf(LINE_FMT, 'Accepts', this.acceptable.join(', '));
     str += sprintf(LINE_FMT, 'Name', this.name);
-    str += sprintf(LINE_FMT, 'Pre', handlersToString(this.before));
-    str += sprintf(LINE_FMT, 'Router', this.router.toString());
+    str += sprintf(
+        LINE_FMT,
+        'Pre',
+        handlersToString(this.preChain.getHandlers())
+    );
+    str += sprintf(LINE_FMT, 'Router', '');
+    this.router
+        .toString()
+        .split('\n')
+        .forEach(function forEach(line) {
+            str += sprintf('\t\t%s\n', line);
+        });
     str += sprintf(LINE_FMT, 'Routes', '');
-    Object.keys(this.routes).forEach(function forEach(k) {
-        var handlers = handlersToString(self.routes[k]);
-        str += sprintf(SUB_LINE_FMT, k, handlers);
+    _.forEach(this.router.getRoutes(), function forEach(route, routeName) {
+        var handlers = handlersToString(route.chain.getHandlers());
+        str += sprintf(SUB_LINE_FMT, routeName, handlers);
     });
     str += sprintf(LINE_FMT, 'Secure', this.secure);
     str += sprintf(LINE_FMT, 'Url', this.url);
-    str += sprintf(
-        LINE_FMT,
-        'Version',
-        Array.isArray(this.versions) ? this.versions.join() : this.versions
-    );
 
     return str;
 };
 
 ///--- Private methods
 
+// Lifecycle:
+//
+// 1. _onRequest (handle new request, setup request and triggers pre)
+//   2.1 _runPre
+//     2.2. _afterPre (runs after pre handlers are finisehd, triggers route)
+//       3.1 _runRoute
+//         3.2. _afterRoute (runs after route handlers are finised,
+//                          triggers use)
+//          4.1 _runUse
+//            4.2. _afterUse
```

Comment:
I know we chatted about this before, are we changing the pre -> use -> route order? Looks like this is breaking from 6.x.

Reply:
No, it's the same, the doc was wrong, fixed.